packages/foundation-models/src/openai/openai-embedding-response.ts

```diff
@@ -0,0 +1,38 @@
+import { HttpResponse } from '@sap-cloud-sdk/http-client';
+import { createLogger } from '@sap-cloud-sdk/util';
+import { OpenAiEmbeddingOutput } from './openai-types.js';
+import { openAiEmbeddingOutputSchema } from './openai-types-schema.js';
+
+const logger = createLogger({
+  package: 'gen-ai-hub',
+  messageContext: 'openai-embedding-response'
+});
+
+/**
+ * OpenAI embedding response.
+ */
+export class OpenAiEmbeddingResponse {
+  /**
+   * The embedding response.
+   */
+  public readonly data: OpenAiEmbeddingOutput;
+  constructor(public readonly rawResponse: HttpResponse) {
```

Comment:
[q] Oh, so now we actually do not check this in our tests, but on every user request?
[pp] I expected this to just be a test, running this every time seems too much to me. At least it should be optional...

Reply:
I can also remove this check here and put some in the tests. Not sure how much performance it will improve, but now I see a potential benefit if I agree with you. So in case our schema is not 100% in sync with openai, without this mandatory check, user can still treat it as any and access properties freely, especially for pure js developer, who probably do not like this schema check. Otherwise, if schema not totally correct, they cannot use the client any more due to the zod error.

So tl;dr, I will move them into test, not a big effort anyways. Thanks for your feedback!

Reply:
Done.